src/status_im2/navigation/core.cljs

```diff
@@ -108,6 +125,10 @@
                      (dissmissModal)
                      (navigation/pop (name @state/root-id)))))
 
+(re-frame/reg-fx :navigate-back-within-stack
+                 (fn [comp-id]
+                   (navigation/pop (name comp-id))))
```

Comment:
oh ok, so it will work for modal stack as well?

Reply:
Yes, it will work for modals, we just need to specify the correct componentId (https://wix.github.io/react-native-navigation/docs/stack-programmatically/#interact-with-the-stack-by-componentid).